distributed/tests/test_batched.py

```diff
@@ -253,3 +254,52 @@ async def test_serializers():
 
         with pytest.raises(TimeoutError):
             msg = await asyncio.wait_for(comm.read(), 0.1)
+
+
+@pytest.mark.asyncio
+async def test_handles_exceptions():
+    # Ensure that we properly handle exceptions in BatchedSend.
+    # https://github.com/pangeo-data/pangeo/issues/788
+    # mentioned in https://github.com/dask/distributed/issues/4080, but
+    # possibly distinct.
+    #
+    # The reported issues (https://github.com/tornadoweb/tornado/pull/2008)
+    # claim that the BufferError *should* only happen when the application
+    # is incorrectly using threads. I haven't been able to construct an
```

Comment:
https://github.com/tornadoweb/tornado/pull/2008#issuecomment-312982780 may actually be relevant here. stackimpact was calling `sys._current_frames()`. I wonder if our profiler might be causing similar issues? Let me try playing with that.